### PR TITLE
Added indexes to ensure unique key props

### LINKED ISSUE #150: Fix frontend tests warning
CI/CD tests run green but produce the following warning:

```
PASS  __tests__/src/pages/Projects.test.tsx (5.078 s)
  ● Console
    console.error
      Warning: Each child in a list should have a unique "key" prop.
      
      Check the render method of `Card`. See https://reactjs.org/link/warning-keys for more information.
          at ContributorAvatar (/home/runner/work/Nest/Nest/frontend/src/components/ContributorAvatar.tsx:4:30)
          at Card (/home/runner/work/Nest/Nest/frontend/src/components/Card.tsx:14:3)
          at div
          at div
          at Projects (/home/runner/work/Nest/Nest/frontend/src/pages/Projects.tsx:10:49)

      84 |           ))}
      85 |       </h2>
    > 86 |       <div className=" w-full flex justify-normal items-center gap-1 ">
         |       ^
      87 |         {topContributors &&
      88 |           topContributors.map((contributor) => (
      89 |             <ContributorAvatar key={contributor.login} contributor={contributor} />

      at printWarning (node_modules/react/cjs/react-jsx-runtime.development.js:87:30)
      at error (node_modules/react/cjs/react-jsx-runtime.development.js:61:7)
      at validateExplicitKey (node_modules/react/cjs/react-jsx-runtime.development.js:1078:5)
      at validateChildKeys (node_modules/react/cjs/react-jsx-runtime.development.js:1105:11)
      at jsxWithValidation (node_modules/react/cjs/react-jsx-runtime.development.js:1277:11)
      at jsxWithValidationDynamic (node_modules/react/cjs/react-jsx-runtime.development.js:1320:12)
      at Card (src/components/Card.tsx:86:7)
      at renderWithHooks (node_modules/react-dom/cjs/react-dom.development.js:15486:18)
      at mountIndeterminateComponent (node_modules/react-dom/cjs/react-dom.development.js:20103:13)
      at beginWork (node_modules/react-dom/cjs/react-dom.development.js:21626:16)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:27465:14)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:26599:12)
      at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:26505:5)
      at renderRootSync (node_modules/react-dom/cjs/react-dom.development.js:26473:7)
      at performConcurrentWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:25777:74)
      at workLoop (node_modules/scheduler/cjs/scheduler.development.js:266:34)
      at flushWork (node_modules/scheduler/cjs/scheduler.development.js:239:14)
      at performWorkUntilDeadline (node_modules/scheduler/cjs/scheduler.development.js:533:21)
      at Timeout.task [as _onTimeout] (node_modules/jsdom/lib/jsdom/browser/Window.js:520:19)
Test Suites: 3 passed, 3 total
Tests:       4 passed, 4 total
```

Source: https://github.com/OWASP/Nest/actions/runs/12215821212/job/34078260621#step:8:25

It seems we need to specify `key` properly somewhere.

### PR DESCRIPTION
**Pull request by yashpandey06**
Resolves #150 

Modified Card and Project component a bit to resolve errors thrown in test job in pipeline

### CODE REVIEW

File: frontend/src/components/Card.tsx [hunk 1]
```diff
@@ -44,129 +44,124 @@ const Card = ({
...
-        {leaders &&
-          leaders.map((leader, index) => (
-            <span key={leader} className=" text-gray-600 dark:text-gray-300">
-              {' '}
-              {index != leaders.length - 1 ? `${leader},` : `${leader}`}{' '}
+          )}
+          {leaders &&
+              leaders.map((leader, index) => (
+                  <span key={`${leader}-${index}`} className="text-gray-600 dark:text-gray-300">
+              {index !== leaders.length - 1 ? `${leader},` : `${leader}`}
```
Comment (arkid15r) [inline frontend/src/components/Card.tsx]:
```suggestion
              {index !== leaders.length - 1 ? `${leader}, ` : `${leader}`}
```

File: .github/workflows/ci-cd.yaml [hunk 2]
```diff
@@ -86,6 +86,13 @@ jobs:
         with:
           languages: ${{ matrix.language }}
 
+      - name: Set up Node
+        uses: actions/setup-node@v4
```
Comment (arkid15r) [inline .github/workflows/ci-cd.yaml]:
```suggestion
      - name: Set up Node
        if: matrix.language == 'javascript-typescript'
        uses: actions/setup-node@v4
```

**Review by arkid15r — APPROVED** [review]
Thanks for fixing that! The error message is gone thanks to your refactoring!

**Comment by yashpandey06** [comment]
@arkid15r , yeah .. we are clear with this now .

I ll focus on other assigned issues now !

**Review by arkid15r — APPROVED** [review]

**Pull request merged**